packages/grid/x-data-grid/src/colDef/gridBooleanOperators.ts

```diff
@@ -1,20 +1,21 @@
 import { GridFilterInputBoolean } from '../components/panel/filterPanel/GridFilterInputBoolean';
 import { GridFilterItem } from '../models/gridFilterItem';
 import { GridFilterOperator } from '../models/gridFilterOperator';
+import { v7 } from './utils';
 
 export const getGridBooleanOperators = (): GridFilterOperator<any, boolean | null, any>[] => [
   {
     value: 'is',
-    getApplyFilterFn: (filterItem: GridFilterItem) => {
+    getApplyFilterFn: v7((filterItem: GridFilterItem) => {
       if (!filterItem.value) {
         return null;
       }
 
       const valueAsBoolean = filterItem.value === 'true';
-      return ({ value }): boolean => {
+      return (value, _, __, ___): boolean => {
```

Comment:
In the context of https://github.com/mui/mui-x/pull/9254#discussion_r1224584563, I just realized that the only reason TS inferred the arguments properly here is the number of the argument.
Removing `_, __, ___` breaks it 😅

<img width="512" src="https://github.com/mui/mui-x/assets/13808724/8dd95f32-6fb4-43b6-93d5-933cb6c9b655">